contracts/pcv/sentinel/PCVSentinel.sol

```diff
@@ -0,0 +1,173 @@
+    }
+
+    // ---------- Read-Only API ----------
+
+    /**
+     * @notice returns whether or not the given address is a guard
+     * @param guard the address to check
+     */
+    function isGuard(address guard) 
+        external 
+        view
+        override
+        returns (bool) 
+    {
+        return guards.contains(guard);
+    }
+
+    /**
+     * @notice returns a list of all guards
+     */
+    function allGuards() 
+        external
+        view
+        override
+        returns (address[] memory all) 
+    {
+        all = new address[](guards.length());
+
+        for (uint i = 0; i < guards.length(); i++) {
+            all[i] = guards.at(i);
+        }
+
+        return all;
+    }
+
+    /**
+     * @notice returns the list of all guards that pass their check
+     * @dev you *can* use this on-chain if you want, but this is made
+     * to be called off-chain to pass into protecManys and not hyper-efficient
+     */
+    function checkAll()
+        external 
+        view
+        override 
+        returns (address[] memory) 
+    {
+        uint256 numGuardsCanProtec;
+
+        for (uint256 i=0; i<guards.length(); i++) {
+            if(IGuard(guards.at(i)).check()) {
+                numGuardsCanProtec++;
+            }
+        }
+
+        address[] memory guardsWhoCanProtec = new address[](numGuardsCanProtec);
+
+        for (uint256 i=0; i<guards.length(); i++) {
+            if(IGuard(guards.at(i)).check()) {
+                numGuardsCanProtec--;
+                guardsWhoCanProtec[numGuardsCanProtec] = guards.at(i);
+            }
+        }
+
+        return guardsWhoCanProtec;
+    }
+
+    // ---------- Governor-or-Admin-Or-Guardian-Only State-Changing API ----------
+
+    /**
+     * @notice adds a guard contract to the PCV Sentinel
+     * @param guard the guard-contract to add
+     */
+    function knight(address guard) 
+        external 
+        override 
+        isGovernorOrGuardianOrAdmin
+    {
+        guards.add(guard);
+
+        // Inform the kingdom of this glorious news
+        emit GuardAdded(guard);
+    }
+
+    /**
+     * @notice removes a guard
+     * @param traitor the guard-contract to remove
+     */
+    function slay(address traitor) 
+        external 
+        override 
+        isGovernorOrGuardianOrAdmin
+    {
+        guards.remove(traitor);
+
+        // Inform the kingdom of this sudden and inevitable betrayal
+        emit GuardRemoved(traitor);
+    }
+
+    // ---------- Public State-Changing API ----------
+
+    /**
+     * @notice checks the guard and runs its protec actions if needed
+     * @param guard the guard to activate
+     */
+    function protec(address guard) 
+        external
+        override
+    {
+        require(guards.contains(guard), "Guard does not exist.");
+
+        bool needsProtec = IGuard(guard).check();
+
+        if (needsProtec) {
+            (address[] memory targets, bytes[] memory datas) = IGuard(guard).getProtecActions();
```

Comment:
"data" has no plural

Reply:
technically you are correct, but I add an s to help differentiate an array of data-objects from a single data object

Reply:
calldatas?